napari/layers/base/base.py

```diff
@@ -602,8 +614,15 @@ def shape(self):
         -------
         shape : tuple
         """
-        # To Do: Deprecate when full world coordinate refactor
-        # is complete
+        warnings.warn(
+            (
+                "The shape parameter is deprecated and will be removed in version 0.4.1."
+                " Instead you should use the extent.data and extent.world parameters"
+                " to get the extent of the data in data or world coordinates."
+            ),
+            category=DeprecationWarning,
```

Comment:
This should (counter-intuitively, I admit) be a FutureWarning, rather than a DeprecationWarning. The latter are not visible by default.

Reply:
```suggestion
            category=FutureWarning,
```